Sources/CoreMetrics/Metrics.swift

```diff
@@ -787,6 +817,24 @@ internal final class AccumulatingMeter: MeterHandler {
     }
 
     func increment(by amount: Double) {
+        // Drop illegal values
+        // - cannot increment by NaN
+        guard !amount.isNaN else {
+            return
+        }
+        // - cannot increment by infinite quantities
+        guard !amount.isInfinite else {
+            return
+        }
+        // - cannot increment by negative values
+        guard amount.sign == .plus else {
+            return
+        }
+        // - cannot increment by zero
+        guard !amount.isZero else {
+            return
+        }
```

Comment:
Should we just silently return here or give the user any signal that he used an invalid value?

Reply:
we could make this throwing, but not sure if that make the call sites more cumbersome, note this is the default implementation, so in most cases the actual backend will drive the behavior

Reply:
I don't throwing makes sense here since it makes the API very awkward to use. I was more thinking if we should precondition this but that's probably also not a great idea. Overall, I am fine with just dropping them on the ground but wanted to raise the question.

cc @ktoso @Lukasa @fabianfett

Reply:
We could `assert` on this to ensure that debug builds fail.

Reply:
that isa. good idea